File: need4spd/euler_39.pl
my %r=();

foreach my $a ((1..1000)) {
    foreach my $b ((1..1000)) {
        foreach my $c ((1..1000)) {
            
            if ($a + $b + $c > 1000) {
                last;
            }
            
            my $temp = $a**2 + $b**2;
            if ($c**2 == $temp) {
                my $p = $a + $b + $c;
                
                if ($r{$p}) {
                    $r{$p} = $r{$p} + 1;
                } else {
                    $r{$p} = 1;
                }
            }
            
        }
    }
}

@max_keys = sort {$r{$b} <=> $r{$a}} keys %r;

print "@max_keys \n";
print $max_keys[0];

File: need4spd/euler_39.rb
r=Hash.new()

(1..1000).each do |a|
  (1..1000).each do |b|
    (1..1000).each do |c|
      
      if (a+b+c > 1000)
        break
      end
      
      temp = a**2 + b**2
      
      if (c**2 == temp)
        p=a+b+c
        
        if (r.has_key? p)
          r[p] = r[p]+1
        else
          r[p] = 1
        end
      end
    end
  end
end

#sorted_r = r.sort {|a, b| a[1] <=> b[1]}
#puts sorted_r

#puts "###################"

sorted_r = r.sort {|a, b| -(a[1] <=> b[1])}
puts sorted_r

